dex-it/src/test/scala/com/wavesplatform/it/sync/MultipleMatchersTestSuite.scala

```diff
@@ -67,6 +72,62 @@ class MultipleMatchersTestSuite extends MatcherSuiteBase {
     dex1.connectToNetwork()
   }
 
+  "WS Order book state should be the same on two matchers" in {
+    val acc = mkAccountWithBalance(100.eth -> eth, 100.waves -> Waves)
+
+    val wsob1 = mkWsOrderBookConnection(ethWavesPair, dex1)
+    val wsob2 = mkWsOrderBookConnection(ethWavesPair, dex2)
+
+    val sell     = mkOrder(acc, ethWavesPair, SELL, 10.eth, 1.waves, 0.003.waves)
+    val buyPart1 = mkOrder(alice, ethWavesPair, BUY, 5.eth, 1.waves, 0.003.waves)
+    val buyPart2 = mkOrder(alice, ethWavesPair, BUY, 3.eth, 1.waves, 0.003.waves)
+    val buyPart3 = mkOrder(alice, ethWavesPair, BUY, 2.eth, 1.waves, 0.003.waves)
+
+    dex1.api.place(sell)
+    dex1.api.place(buyPart1)
+    dex1.api.place(buyPart2)
+    dex1.api.place(buyPart3)
+
+    val obs1 = wsob1.messages
+      .reduce((m1, m2) => mergeOrderBookChanges(m1.asInstanceOf[WsOrderBookChanges], m2.asInstanceOf[WsOrderBookChanges]))
+
+    val obs2 = wsob2.messages
+      .reduce((m1, m2) => mergeOrderBookChanges(m1.asInstanceOf[WsOrderBookChanges], m2.asInstanceOf[WsOrderBookChanges]))
+
+    obs1 should be equals obs2
```

Comment:
1. Not only `WsOrderBookChanges` comes from the server
2. In Scala you should have a strong reason to use `asInstanceOf`, because you probably miss other cases ^ and make a possible error a runtime error instead of a compile time one
3. Also an update of order book couldn't come. It makes this test flaky

Suggest:
1. Wait `sell` order to be filled before reducing messages
2. Use `receiveAtLeastN[WsOrderBookChanges](1)` to get only `WsOrderBookChanges`
3. Wrap receiving and assertion in `eventually`, because all messages could not come instantly even we have waiting for `sell` to be filled

Reply:
ok